server/src/test/java/io/unitycatalog/server/base/catalog/BaseCatalogCRUDTest.java

```diff
@@ -47,24 +47,38 @@ public void testCatalogCRUD() throws ApiException {
     CatalogInfo catalogInfo = catalogOperations.createCatalog(createCatalog);
     assertCatalog(catalogInfo, CATALOG_NAME, COMMENT, PROPERTIES);
 
+    // Create another catalog to test pagination
+    CreateCatalog createCatalog2 =
+        new CreateCatalog().name(COMMON_ENTITY_NAME).comment(COMMENT).properties(PROPERTIES);
+    CatalogInfo catalogInfo2 = catalogOperations.createCatalog(createCatalog2);
+    assertCatalog(catalogInfo2, COMMON_ENTITY_NAME, COMMENT, PROPERTIES);
+
     // List catalogs
     System.out.println("Testing list catalogs..");
-    List<CatalogInfo> catalogList = catalogOperations.listCatalogs();
+    List<CatalogInfo> catalogList = catalogOperations.listCatalogs(Optional.empty());
     assertThat(catalogList).isNotNull();
     assertCatalogExists(catalogList, CATALOG_NAME, COMMENT, PROPERTIES);
+    assertCatalogExists(catalogList, COMMON_ENTITY_NAME, COMMENT, PROPERTIES);
+
+    // List catalogs with page token
+    System.out.println("Testing list catalogs with page token..");
+    catalogList = catalogOperations.listCatalogs(Optional.of(CATALOG_NAME));
```

Comment:
Does this test implicitly assume that the page token is the catalog name? The page token is intended to be opaque with the client not interpreting it. Can you rewrite the test to do the previous `listCatalogs` call with max result of 1 and use the returned page token to do another `listCatalogs` call?

Reply:
The page token being used is the [name of the last entity](https://github.com/unitycatalog/unitycatalog/blob/141936063f3aae6aeb0185110b0731a7059ec64a/server/src/main/java/io/unitycatalog/server/persist/utils/PagedListingHelper.java#L61) of the page and is contained in the [properties of `ListCatalogsResponse`](https://github.com/unitycatalog/unitycatalog/blob/141936063f3aae6aeb0185110b0731a7059ec64a/api/Models/ListCatalogsResponse.md) (and equivalent responses for other entities). But, the [tests are using `List<CatalogInfo>`](https://github.com/unitycatalog/unitycatalog/blob/141936063f3aae6aeb0185110b0731a7059ec64a/server/src/test/java/io/unitycatalog/server/base/catalog/CatalogOperations.java#L13) which, as expected, does not contain any information regarding the `next_page_token` - so for brevity (as the PR is already big) I used the entity name directly. 

I can make the changes and replace `List<CatalogInfo>` with `ListCatalogsResponse` (and do the same for all other entities) in this same PR or if necessary, I can create a new PR for these changes.

Please let me know if I am missing something or if we should continue with the current approach.

Thanks!

Reply:
Thanks! It is ok to do it in a new PR.